compiler/src/dotty/tools/dotc/typer/QuotesAndSplices.scala

```diff
@@ -45,10 +45,16 @@ trait QuotesAndSplices {
         ctx.warning("Canceled splice directly inside a quote. '[ ${ XYZ } ] is equivalent to XYZ.", tree.sourcePos)
       case _ =>
     }
+    val qctx = inferImplicitArg(defn.QuoteContextClass.typeRef, tree.span)
+    if (level == 0 && qctx.tpe.isInstanceOf[SearchFailureType])
+      ctx.error(missingArgMsg(qctx, defn.QuoteContextClass.typeRef, ""), ctx.source.atSpan(tree.span))
     val tree1 =
-      if (ctx.mode.is(Mode.Pattern) && level == 0) typedQuotePattern(tree, pt)
-      else if (tree.quoted.isType) typedTypeApply(untpd.TypeApply(untpd.ref(defn.InternalQuoted_typeQuote.termRef), tree.quoted :: Nil), pt)(quoteContext)
-      else typedApply(untpd.Apply(untpd.ref(defn.InternalQuoted_exprQuote.termRef), tree.quoted), pt)(quoteContext)
+      if ctx.mode.is(Mode.Pattern) && level == 0 then
+        typedQuotePattern(tree, pt, qctx)
+      else if (tree.quoted.isType)
+        typedTypeApply(untpd.TypeApply(untpd.ref(defn.InternalQuoted_typeQuote.termRef), tree.quoted :: Nil), pt)(quoteContext)
+      else
+        typedApply(untpd.Apply(untpd.ref(defn.InternalQuoted_exprQuote.termRef), tree.quoted), pt)(quoteContext).select(nme.apply).appliedTo(qctx)
```

Comment:
I'm not sure about this line, as previously, we don't need to `.select(nme.apply).appliedTo(qctx)`, and it worked.

Reply:
Previously we were letting typer find and apply it. But as we know it we can just add it explicitly. In the previous case, it did not find the implicit it emitted an error mentioning `CompileTime.quotedExpr` in some situations.